python-packages/presto_drf/presto_drf/mixins.py

```diff
@@ -0,0 +1,65 @@
+class SerializerOptInFieldsMixin:
+    """
+    Regulates fields exposed by default & as requested based on query parameters or context.
+
+    Pass 'include_fields' / 'opt_in_fields' thru query params or context to use; multiple fields are separated by comma eg,
+    /?include_fields=first_name,email&opt_in_fields=gait_recognition_prediction
+
+    1. By default, all "fields" defined in serializer, minus those listed in "opt_in_fields" would be returned.
+    2. If "include_fields" is supplied, only fields requested this way would be returned.
+    3. If "opt_in_fields" is supplied, fields requested this way PLUS fields from #1 or #2 would be returned.
+
+    Pinned fields are always returned. (currently pk only if pk exists)
+    """
+
+    def get_pinned_fields(self):
```

Comment:
can you add header comment that will show up in docs + return type?

Reply:
done